pkg/mcp/client/client.go

```diff
@@ -261,6 +261,11 @@ func (c *Client) Run(ctx context.Context) {
 				}
 			} else {
 				collection := req.TypeUrl
+
+				if req.ErrorDetail != nil {
+					break
+				}
+
```

Comment:
This will leak streams. This exits the message loop but doesn't close current stream and a new stream will be re-created (see lines 196-215). 

To rate limit ACK/NACK on the client we don't need to modify the stream. I think simply adding an inline sleep (w/select over ctx and time.After) should be sufficient.

Also, as @ozevren said, this should be investigated and fixed in the new MCP source/sink stack.